### PR TITLE
[UI tests] implement correct waitTillPageIsLoaded on various pages

### LINKED ISSUE #33161: Acceptance test OwncloudPage and waitTillPageIsLoaded
See PR #33158 and #33149 for examples and discussion.

OwncloudPage has a ``waitTillPageIsLoaded()`` method that looks for a ``.loading`` indicator to become invisible. On ordinary files pages there is no such indicator, and FilesPageBasic extends OwncloudPage and implements a ``waitTillPageIsLoaded()`` that works on files-type-pages.

Some other pages directly extend OwncloudPage and so get the "base" ``waitTillPageIsLoaded()``

When the comments app is enabled, the comments panel happens to have a ``.loading`` indicator - so the "base" ``waitTillPageIsLoaded()`` just happens to "work" by checking that page element.

- [ ] review each page object that extends OwncloudPage and ensure that either it implements its own appropriate ``waitTillPageIsLoaded()`` or that the base one from OwncloudPage is relevant.

### PR DESCRIPTION
**Pull request by individual-it**
## Description
- introduce correct `waitTillPageIsLoaded()` function for every page where it makes sense
- use this waiting consequently

## Related Issue

- Fixes #33161

## Motivation and Context
be independent of "random" waiting and wait for the page the show up correctly

## How Has This Been Tested?
:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/lib/OwncloudPage.php [hunk 1]
```diff
@@ -143,6 +143,41 @@ public function waitTillElementIsNotNull(
...
+		Session $session,
+		$xpath,
+		$timeout_msec = STANDARD_UI_WAIT_TIMEOUT_MILLISEC
+	) {
+		$element = $this->waitTillElementIsNotNull($xpath);
+		$this->assertElementNotNull(
+			$element,
+			__METHOD__ .
+			" xpath: $xpath" .
+			" timeout waiting for element to be availiable"
```
Comment (phil-davis) [inline tests/acceptance/features/lib/OwncloudPage.php]: s/availiable/available/

File: tests/acceptance/features/lib/OwncloudPage.php [hunk 2]
```diff
@@ -143,6 +143,41 @@ public function waitTillElementIsNotNull(
...
+		);
+		$visibibity = $this->waitFor(
```
Comment (phil-davis) [inline tests/acceptance/features/lib/OwncloudPage.php]: s/visibibity/visibility

**Review by phil-davis — APPROVED** [review]
Looks great, assuming CI passes!

I only noticed 2 little typos - fix those if you care, but they are not critical.

**Comment by individual-it** [comment]
backport in #33589 

> I only noticed 2 little typos - fix those if you care, but they are not critical.

Usually I would care but as drone is struggling today and the issues are not functional I care more about getting these changes into the QA build at night

**Pull request merged**